src/sql/workbench/contrib/query/browser/queryActions.ts

```diff
@@ -663,6 +667,10 @@ export class ListDatabasesActionItem extends Disposable implements IActionViewIt
 			return;
 		}
 
+		if (dbName === this.getCurrentDatabaseName()) {
```

Comment:
Would suggest putting this at the top, right below the !dbName check so that we avoid having to call getConnectionProfile if we can.

Reply:
@Charles-Gagnon, moved the check below the !dbName check to avoid calling getConnectionProfile.